.github/workflows/build_test_package.yaml

```diff
@@ -0,0 +1,321 @@
+      - name: Run RBAC Workspace integration tests
+        run: |
+          ./gradlew :cosmotech-workspace-api:integrationTest --tests "com.cosmotech.workspace.service.WorkspaceServiceRBACTest" -x spotlessCheck -x openApiValidate
+  runner-it:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    needs:
+      - test-unit
+    permissions: read-all
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: "21"
+      - name: Run Runner integration tests
+        run: |
+          ./gradlew :cosmotech-runner-api:integrationTest --tests "com.cosmotech.runner.service.RunnerServiceIntegrationTest" -x spotlessCheck -x openApiValidate
+      - name: Run RBAC Runner integration tests
+        run: |
+          ./gradlew :cosmotech-runner-api:integrationTest --tests "com.cosmotech.runner.service.RunnerServiceRBACTest" -x spotlessCheck -x openApiValidate
+  run-it:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    needs:
+      - test-unit
+    permissions: read-all
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: "21"
+      - name: Launch Run integration tests
+        run: |
+          ./gradlew :cosmotech-run-api:integrationTest -x spotlessCheck -x openApiValidate
+  homecontroller-it:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    needs:
+      - test-unit
+    permissions: read-all
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: "21"
+      - name: Run HomeController integration tests
+        run: |
+          ./gradlew :cosmotech-api:integrationTest -x spotlessCheck -x openApiValidate
+  scan-and-push-container:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_BUILDKIT: 1
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    needs:
+      - test-unit
+      - connector-it
+      - dataset-it
+      - organization-it
+      - solution-it
+      - workspace-it
+      - runner-it
+      - run-it
+      - homecontroller-it
+    permissions:
+      packages: write
+      contents: read
+      security-events: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Fetch all tags since Gradle project version is built upon SCM
+          fetch-depth: 0
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: "21"
+
+      - name: retrieve tag name
+        id: refenv
+        uses: Cosmo-Tech/cosmotech-api/.github/actions/retrievetag@ndon/argocd
```

Comment:
This should not reference a working branch. Maybe we can extract the retrievetag code as you did, but if we only use it once here, maybe inlining the code is good enough.